pkg/controller/history/controller_history.go

```diff
@@ -454,3 +452,31 @@ func (fh *fakeHistory) ReleaseControllerRevision(parent metav1.Object, revision
 	}
 	return clone, fh.indexer.Update(clone)
 }
+
+func getPodLabelsFromData(data runtime.RawExtension) map[string]string {
+	var dat map[string]interface{}
+	labelMap := make(map[string]string)
+	if err := json.Unmarshal(data.Raw, &dat); err != nil {
+		return labelMap
+	}
+	spec, ok := dat["spec"].(map[string]interface{})
+	if !ok {
+		return labelMap
+	}
+	template, ok := spec["template"].(map[string]interface{})
+	if !ok {
+		return labelMap
+	}
+	metadata, ok := template["metadata"].(map[string]interface{})
+	if !ok {
+		return labelMap
+	}
+	labels, ok := metadata["labels"].(map[string]interface{})
+	if !ok {
+		return labelMap
+	}
+	for key, value := range labels {
+		labelMap[key] = value.(string)
```

Comment:
We should make sure to validate both the key and value to make sure they are valid label key and label values.

Reply:
Are we required to validate key and values again? Because they must have already been validated when creating the StatefulSet (or pods).